### PR TITLE
Fix vs2017 build

### PR DESCRIPTION
**Pull request by guparan**
This is a bunch of changes I had to do to be able to configure and build SofaPython3 on my machine.

I am not sure about the last commit. What is the point of `install(EXPORT ...` on tests?

### CODE REVIEW

**Comment by jnbrunet** [comment]
> What is the point of `install(EXPORT ...` on tests?

None I guess, probably a bad copy/paste on my part. The only important thing is the `install(DIRECTORY ...)` as we package those for the CI (tests are done on an installed SP3, not compiled).

**Comment by jnbrunet** [comment]
Hum, the CI cannot find the tests anymore. I guess you do need the install(EXPORT ...).

File: CMakeLists.txt [hunk 1]
```diff
@@ -60,6 +60,15 @@ if (NOT SP3_COMPILED_AS_SUBPROJECT)
     set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${LIBRARY_OUTPUT_DIRECTORY})
 endif()
 
+if(MSVC)
+    # define BOOST_ALL_DYN_LINK needed for dynamic linking with boost libraries
+    add_definitions(-DBOOST_ALL_DYN_LINK)
```
Comment (jnbrunet) [inline CMakeLists.txt:65]: Not needed anymore, see sofa-framework/sofa#1578

**Comment by guparan** [comment]
Test failures are not my fault anymore! :angel:

**Comment by guparan** [comment]
Please merge me! :pray:

**Pull request merged**